include/lsquic.h

```diff
@@ -1186,6 +1186,12 @@ struct lsquic_engine_api
      */
     const struct lsquic_packout_mem_if  *ea_pmi;
     void                                *ea_pmi_ctx;
+
+    /**
+     * Optional interface to control the creation of connection IDs
+     */
+    void  (*ea_generate_cid)(lsquic_cid_t *);
```

Comment:
Was this a conscious decision on your part not to include `lsquic_conn_t *` in the list of arguments?  You may not need it in your use case, but having some way to get back at the context in the callback may be useful to others.

We should rename the function to `es_generate_scid` to indicate that these are *source* CIDs that are being generated.

Reply:
makes sense. done!